desktop-widgets/importgps.cpp

```diff
@@ -0,0 +1,329 @@
+		// == get time: ==
+		if (find_xml_element(f1, "time", buf, '>')) // Find the <time> element
+			break; // on EOF
+		if (get_substring(f1, buf, '<'))  // Read the string containing date/time
+			break; // on EOF
+		tm1.tm_year = atoi(substr(buf, strbuf, 0,3));  // Extract the different substrings:
+		tm1.tm_mon = atoi(substr(buf, strbuf, 5,6))-1;
+		tm1.tm_mday = atoi(substr(buf, strbuf, 8,9));
+		tm1.tm_hour = atoi(substr(buf, strbuf, 11,12));
+		tm1.tm_min = atoi(substr(buf, strbuf, 14,15));
+		tm1.tm_sec = atoi(substr(buf, strbuf, 17,18));
+		when = utc_mktime(&tm1) + time_offset;
+		if (first_line) {
+			first_line = false;
+			coords.start_track = when;
+		}
+		if ((when > divetime) && (found == false)) {   // This GPS time corresponds to the start of the dive
+			coords.lon = lon; // save the coordinates
+			coords.lat = lat;
+			found = true;
+		}
+#ifdef GPSDEBUG
+		utc_mkdate(when, &time); // print time and coordinates of each of the trkpt elements of the GPX file
+		fprintf(stderr, " %02d: lat=%f lon=%f timestamp=%ld (%ld) %02d/%02d/%02d %02d:%02d  dt=%ld  %02d/%02d/%02d %02d:%02d\n", line, lat,
+		lon, when, time_offset, time.tm_year, time.tm_mon+1, time.tm_mday, time.tm_hour, time.tm_min, divetime, dyr, dmon+1, dday,dhr, dmin );
+#endif
+	} while (true); // This loop executes until EOF causes a break out of the loop
+	coords.end_track = when;
+	fclose(f1);
+	return 0;
+}
+
+// Fill the visual elements of the synchronisation panel with information
+void  ImportGPS::updateUI()
+{
+	struct tm time;
+	int dive_day, gps_day;
+	char datestr[50];
+	QString problemString;
+
+	utc_mkdate(coords.start_track, &time); // Display GPS date and start and end times:
+	gps_day = time.tm_mday;
+	datestr[0] = 0x0;
+	strftime(datestr, sizeof(datestr), "%A %d %B ", &time); // GPS date
+	ui.trackDateLabel->setText("GPS date  = " + QString(datestr) + QString::number(time.tm_year));
+	ui.startTimeLabel->setText(QString::number(time.tm_hour) + ":" + QString::number(time.tm_min)); // track start time
+	utc_mkdate(coords.end_track, &time);
+	ui.endTimeLabel->setText(QString::number(time.tm_hour) + ":" + QString::number(time.tm_min));  // track end time
+
+	utc_mkdate(coords.start_dive, &time); // Display dive date and start and end times:
+	dive_day = time.tm_mday;
+	datestr[0] = 0x0;
+	strftime(datestr, sizeof(datestr), "%A %d %B ", localtime(&(coords.start_dive))); // dive date
+	ui.diveDateLabel->setText("Dive date = " + QString(datestr) + QString::number(time.tm_year));
+	ui.startTimeSyncLabel->setText( QString::number(time.tm_hour) + ":" + QString::number(time.tm_min)); // dive start time
+	utc_mkdate(coords.end_dive, &time);
+	ui.endTimeSyncLabel->setText(QString::number(time.tm_hour) + ":" + QString::number(time.tm_min));  // dive end time
+
+	// This section implements extensive warnings in case there is not synchronisation between dive and GPS data:
+	if (gps_day != dive_day)
+		problemString = "(different dates)";
+	else 
+		problemString = "";
```

Comment:
can be `problemString = (gps_day != dive_day) ?  .....`

Reply:
Or, perhaps even simpler remove the else branch. `problemString` was initialized to the empty string anyway, no?

Reply:
done.